games/ludii/jni_utils.cc

```diff
@@ -30,17 +30,24 @@ SOFTWARE.
 namespace Ludii {
 
 JavaVM* JNIUtils::jvm = nullptr;
-JNIEnv* JNIUtils::env = nullptr;
 jint JNIUtils::res = 0;
 
 JNIEnv* JNIUtils::GetEnv() {
+  if (jvm == nullptr)
+    return nullptr;
+
+  JavaVMAttachArgs args = {JNI_VERSION_1_2, 0, 0};
+  JNIEnv* env;
+  jvm->AttachCurrentThread((void**)&env, &args);
```

Comment:
Is this the slow method of retrieving the Java env for the thread? This could be stored in a thread local variable

Reply:
Yes. I'll change JNIUtils to have a thread_local env pointer. With that, calls to JNIUtils::GetEnv() should become really fast, so I think I'll change LudiiGameWrapper and LudiiStateWrapper to just ALWAYS call that to get a pointer to their env whenever they need one (rather than the current implementation where I tried to sometimes store it where it "seemed" safe, and sometimes retrieve it again through the expensive method in cases where it seemed necessary).